app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  def create
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    if liked?
+      redirect_to user_posts_path(@user.id)
+    else
+      @like = @user.likes.new
+      @like.author_id = @user.id
+      @like.post_id = @post.id
+      @like.save
+    end
+    redirect_to user_posts_path(@user.id)
+  end
```

Comment:
- [ ] In the posts index page, when I click on the like button, it redirects me to the post-show page rather than creating a like. Please ensure that a user is able to add a like to posts on the posts#index page.